### PR TITLE
Allow expected autoscale capacity

### PR DESCRIPTION
**Pull request by djsd123**
Modify InstancePool count function

```release-note
enable minCount and maxCount of stateless instancePools
```

### CODE REVIEW

**Comment by simonswine** [comment]
@djsd123 thanks for your contribution. I have renamed count to min_count where ever applicable, we need to validate these values properly, otherwise it takes a while to see the terraform error.

File: terraform/amazon/templates/instance_pools/instance_pool_instance.tf.template [hunk 1]
```diff
@@ -124,6 +123,15 @@ resource "aws_autoscaling_group" "{{.TFName}}" {
     propagate_at_launch = true
   }
 }
+
+resource "aws_autoscaling_policy" "{{.TFName}}-autoscaling-policy" {
```
Comment (simonswine) [inline terraform/amazon/templates/instance_pools/instance_pool_instance.tf.template]: As far as I see is the scaling policy doing exactly nothing, let me know if that is not true? I would merge this after we remove the scaling policy...

**Comment by munnerz** [comment]
How does adjusting these values play into upgrades and/or automatically
draining of nodes upon the max_count var being reduce to less than the
current number of nodes in the ASG?

File: terraform/amazon/templates/instance_pools/instance_pool_instance.tf.template (hunk 1 above)
Comment (djsd123) [reply terraform/amazon/templates/instance_pools/instance_pool_instance.tf.template]: @simonswine Yep, spot on. It is indeed superfluous.  Removed as requested.

**Comment by djsd123** [comment]
Hi @munnerz, 

Was that question aimed at me?  Not quite sure I follow.

**Comment by djsd123** [comment]
Ah, good question.

**Comment by djsd123** [comment]
@simonswine Would you prefer I closed this?

**Comment by simonswine** [comment]
Merged as part of #267

**Pull request closed without merging**